sdks/python/apache_beam/runners/worker/sdk_worker.py

```diff
@@ -51,7 +53,7 @@ class SdkHarness(object):
 
   def __init__(
       self, control_address, worker_count, credentials=None, worker_id=None,
-      profiler_factory=None):
+      state_cache_size=100, profiler_factory=None):
```

Comment:
Does this mean we can only cache 100 keys? That may be a bit little, but probably it is best left to the user. How about making the default 0 and have that turn the cross bundle caching off?

Reply:
The cache size setting would need to be passed to the entry point (`apache_beam.runners.worker.sdk_worker_main`) as pipeline option, so that in the end-to-end scenario the user can specify it. https://github.com/apache/beam/blob/5b43602109648830522159221a283754ed959afa/sdks/python/apache_beam/runners/worker/sdk_worker_main.py#L121

Reply:
>Does this mean we can only cache 100 keys? 

Yes.

>How about making the default 0 and have that turn the cross bundle caching off?

Was thinking about this as well. Seems fair to default to 0 in the beginning but IMHO we should change the default to cached after experimentation. Would still keep caching enabled for the tests, except for the test suite which explicitly has caching disabled.

>The cache size setting would need to be passed to the entry point as pipeline option

You can read my mind. Just added this in another commit :)